src/Voron/Data/CompactTrees/CompactKey.cs

```diff
@@ -19,20 +19,29 @@ namespace Voron.Data.CompactTrees;
 public sealed unsafe class CompactKey : IDisposable
 {
     public static readonly CompactKey NullInstance = new();
-    
-    private static readonly ArrayPool<byte> StoragePool = ArrayPool<byte>.Shared;
-    private static readonly ArrayPool<long> KeyMappingPool = ArrayPool<long>.Shared;
+
+    private static readonly PerCoreStatic<ArrayPool<byte>> StoragePool = new (ArrayPool<byte>.Create);
 
     private LowLevelTransaction _owner;
 
     private const int MappingTableSize = 64;
     private const int MappingTableMask = MappingTableSize - 1;
 
-    private long[] _keyMappingCache;
-    private ref long KeyMappingCache(int i) => ref _keyMappingCache[i];
-    private ref long KeyMappingCacheIndex(int i) => ref _keyMappingCache[MappingTableSize + i];
+    private struct KeyMappingCache
+    {
+        private fixed long _keyMappingCache[MappingTableSize * 2];
 
+        public long GetMap(int i) => _keyMappingCache[i];
+        public long GetIndex(int i) => _keyMappingCache[MappingTableSize + i];
 
+        public void Set(int i, long map, long index)
+        {
+            _keyMappingCache[i] = map;
+            _keyMappingCache[MappingTableSize + i] = index;
+        }
+    }
+
+    private KeyMappingCache _keyMappingCache;
```

Comment:
This has the side effect of creating (and zeroing) a lot of memory, no?

Reply:
No. That's what `_lastKeyMappingItem` is for. :)

Reply:
I meant, whenever we create an instance.

Reply:
We seldom create instances of those.